projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/asm/syscall/LLVMAMD64Info.java

```
@@ -0,0 +1,136 @@
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.oracle.truffle.llvm.nodes.asm.syscall;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.llvm.nodes.asm.support.LLVMAMD64String;
+import com.oracle.truffle.llvm.runtime.LLVMAddress;
+
+public class LLVMAMD64Info {
+    public static final String sysname;
+    public static final String release;
+    public static final String machine;
+
+    static {
+        sysname = System.getProperty("os.name");
+        release = System.getProperty("os.version");
+        String arch = System.getProperty("os.arch");
+        if (arch.equals("amd64")) {
+            arch = "x86_64";
+        }
+        machine = arch;
+    }
+
+    private static String readFile(String name, String fallback) {
+        try {
+            Path path = Paths.get(name);
+            if (Files.exists(path)) {
+                return new String(Files.readAllBytes(path)).trim();
+            }
+        } catch (Exception e) {
```

also add transferToInterpreter to empty catch clauses